create.go

```diff
@@ -251,7 +251,9 @@ func (r *Resource) computeCreateEventPatches(ctx context.Context, obj interface{
 				RestConfig: restConfig,
 			}
 			k8sClients, err := k8sclient.NewClients(clientsConfig)
-			if err != nil {
+			if tenant.IsAPINotAvailable(err) || k8sclient.IsTimeout(err) {
```

Comment:
This will have the effect of not returning the `context.DeadlineExceeded` error and instead waiting for the next reconciliation loop. This is mainly to reduce noise in logs but also means we don't repeatedly call the not ready API slightly reduce load and network traffic.

Reply:
Are we OK with the fact that the status might be outdated for another reconciliation loop? Also is there anything in the operator's logs that will hint us to what went wrong? Maybe it's worth adding a log line here.

Reply:
FWIS we normally log "canceling resource" in azure-operator to know that the reconciliation for that handler was cancelled.

Reply:
> Are we OK with the fact that the status might be outdated for another reconciliation loop?

We currently wrap our resource handlers with `retryresource.Wrap` which has a [linear backoff of 3 times with 1 second delay](https://github.com/giantswarm/operatorkit/blob/d35c2261fac41910c8dc379fd13dd28c226219f2/pkg/resource/wrapper/retryresource/wrap.go#L27). If the API isn't currently available, it's unlikely that it will be available after those retries in say 3-10 seconds so we will end up waiting for the next reconciliation loop anyways in my opinion.

Reply:
I've added a debug log line so we have more visibility into the execution of this resource. I will go ahead and merge this and ask for your reviews on the release PR in case you think this needs any more changes.